docs/changelog.rst

```diff
@@ -0,0 +1,115 @@
+Changelog
+---------
+
+Upcoming
+'''''''
+- Dropped support for Python 3.7 and 3.8
```

Comment:
```suggestion
- Dropped support for Python 3.7 and 3.8
- Added contributing guide
- Changed project linter from flake8 to ruff and applied ruff formatting throughout the codebase
- Moved changelog to a separate file and added GitHub action contributor reminder
```

Some more details that I think are worth mentioning. Please fix my wording! Thanks :-)